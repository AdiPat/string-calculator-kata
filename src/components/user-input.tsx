import { Button, Input } from "@nextui-org/react";

export function UserInput() {
  return (
    <div className="flex flex-col justify-center items-center mt-4">
      <div
        className="flex flex-col justify-center items-center"
        style={{ width: "70%" }}
      >
        <label className="text-white text-lg mb-4" htmlFor="numbers">
          Enter string to compute sum:
        </label>
        <Input placeholder="Enter comma separated numbers" />
        <Button size="lg" color="primary" className="mt-4">
          Calculate
        </Button>
      </div>
    </div>
  );
}
