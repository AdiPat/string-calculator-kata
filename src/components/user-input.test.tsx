import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { UserInput } from "./user-input";

describe("UserInput Tests", () => {
  it("renders a user input with a label, input and button", async () => {
    const { container } = render(
      <UserInput
        value={""}
        setResult={function (value: number): void {
          throw new Error("Function not implemented.");
        }}
        setValue={function (value: string): void {
          throw new Error("Function not implemented.");
        }}
        setShowResult={function (showResult: boolean): void {
          throw new Error("Function not implemented.");
        }}
        setError={function (error: string): void {
          throw new Error("Function not implemented.");
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders a user input with a label", async () => {
    const { container } = render(
      <UserInput
        value={""}
        setResult={function (value: number): void {
          throw new Error("Function not implemented.");
        }}
        setValue={function (value: string): void {
          throw new Error("Function not implemented.");
        }}
        setShowResult={function (showResult: boolean): void {
          throw new Error("Function not implemented.");
        }}
        setError={function (error: string): void {
          throw new Error("Function not implemented.");
        }}
      />
    );
    const label = screen.getByText(/Enter string to compute sum:/i);
    expect(label).toBeInTheDocument();
  });

  it("renders a user input with a placeholder", async () => {
    const { container } = render(
      <UserInput
        value={""}
        setResult={function (value: number): void {
          throw new Error("Function not implemented.");
        }}
        setValue={function (value: string): void {
          throw new Error("Function not implemented.");
        }}
        setShowResult={function (showResult: boolean): void {
          throw new Error("Function not implemented.");
        }}
        setError={function (error: string): void {
          throw new Error("Function not implemented.");
        }}
      />
    );
    const input = screen.getByPlaceholderText(/Enter comma separated numbers/i);
    expect(input).toBeInTheDocument();
  });

  it("renders a user input with a button", async () => {
    const { container } = render(
      <UserInput
        value={""}
        setResult={function (value: number): void {
          throw new Error("Function not implemented.");
        }}
        setValue={function (value: string): void {
          throw new Error("Function not implemented.");
        }}
        setShowResult={function (showResult: boolean): void {
          throw new Error("Function not implemented.");
        }}
        setError={function (error: string): void {
          throw new Error("Function not implemented.");
        }}
      />
    );
    const button = screen.getByText(/Calculate/i);
    expect(button).toBeInTheDocument();
  });
});
