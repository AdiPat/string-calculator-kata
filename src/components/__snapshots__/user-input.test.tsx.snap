// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInput Tests renders a user input with a label, input and button 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center mt-4"
  >
    <div
      class="flex flex-col justify-center items-center"
      style="width: 70%;"
    >
      <label
        class="text-white text-lg mb-4"
        for="numbers"
      >
        Enter string to compute sum:
      </label>
      <div
        class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
        data-filled="true"
        data-filled-within="true"
        data-slot="base"
      >
        <div
          class="h-full flex flex-col"
          data-slot="main-wrapper"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-10 min-h-10 rounded-medium transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <div
              class="inline-flex w-full items-center h-full box-border"
              data-slot="inner-wrapper"
            >
              <input
                aria-label="Enter comma separated numbers"
                class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small group-data-[has-value=true]:text-default-foreground"
                data-slot="input"
                id="react-aria-:r0:"
                placeholder="Enter comma separated numbers"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-6 min-w-24 h-12 text-medium gap-3 rounded-large [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover mt-4"
        type="button"
      >
        Calculate
      </button>
    </div>
  </div>
</div>
`;
